src/ray/core_worker/core_worker.cc

```diff
@@ -810,7 +810,7 @@ void CoreWorker::RegisterOwnershipInfoAndResolveFuture(
 Status CoreWorker::Put(const RayObject &object,
                        const std::vector<ObjectID> &contained_object_ids,
                        ObjectID *object_id) {
-  *object_id = ObjectID::FromIndex(worker_context_.GetCurrentTaskID(),
+  *object_id = ObjectID::FromIndex(worker_context_.GetCurrentTaskUniqueBytes(),
```

Comment:
A random thought: what if we change the index to contain the run number of this task. This way each re-run of the same task generates deterministic but yet unique IDs. E.g. `ObjectID::FromIndex(GetCurrentTaskID(), GetRunNumber(), GetNextPutIndex())`

Reply:
+1

Reply:
You mean to include the run number in the hash? That sounds good to me. But I don't think we should include it in the actual ID since we'd have to increase the number of bits too.